tests/test_checkpointing_plugins_all.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+# Script used to manually test various combinations of plugins when used with
+# the checkpointing functionality
+set -euo pipefail
+./test_checkpointing_plugins.sh
```

Comment:
can we add this to the unittests so that they are run by the CI? If too difficult because of multiprocessing, we could just add this script to the CI pipeline?

Reply:
Also, shouldn't we check that the results of the checkpointed strategy are the same of the non-checkpointed one?

Reply:
We can add the script to the pipeline, but I'm not sure how to check for consistency of results... maybe using the metrics dictionary returned from "eval"?

Reply:
yes. Using the dictionary is enough. Please test also a couple of different configurations for PeriodicEval to check that serialization does not happen in the middle of an experience training.

Reply:
I added a test script to check if metrics are exactly the same. I tested periodic eval "by hand" 😅